pkg/util/util.go

```diff
@@ -154,3 +155,50 @@ func InArray(needle interface{}, haystack interface{}) (exists bool, index int)
 
 	return
 }
+
+// SetCondition sets a condition on a custom resource's status
+func SetCondition(conditions *[]gcpv1alpha1.Condition, status corev1.ConditionStatus, reason string, message string) error {
```

Comment:
Would be nice to have tests for this function, as it's part of the public interface of util.go.

Reply:
added one for that

Reply:
is one test good enough for this function? Did you cover all the different routes? Cyclomatic complexity is 4, so I would think the number of tests should be somewhere around that?

Reply:
well, I've tested the main functionality here, not all edge cases. For me that test covers the ability to change the `conditions` array properly. What do you mean by _all the different routes_?

Reply:
If you say your test covers everything, I'm fine :)